src/graphql.ts

```diff
@@ -71,9 +69,7 @@ export interface GraphQLArgs {
   typeResolver?: Maybe<GraphQLTypeResolver<any, any>>;
 }
 
-export function graphql(
-  args: GraphQLArgs,
-): Promise<ExecutionResult | AsyncGenerator<AsyncExecutionResult, void, void>> {
+export function graphql(args: GraphQLArgs): Promise<ExecutionResult> {
```

Comment:
I think this was @IvanGoncharov 's idea but I might be misinterpreting. Probably planning to change when not "experimental"?

Reply:
@robrichard Yes, @glasser is right.
The idea is to finally merge it into `main` having an "experiment" prefix to everything.
Changing the prototype of `graphql` makes it a breaking change, so better to avoid it before we are sure that the shape of the response is final.

So this PR has two things inside:
1. Stuff @glasser learned by implementing this feature in ApolloServer (changes to `execute` typings).
2. Moving everything under the "experiment" prefix and cutting non-essential functionality (e.g. `graphql`).